libslide/src/scanner/types.rs

```diff
@@ -29,66 +32,88 @@ pub enum TokenType{
     Exp,
 
     // Identifies an equal sign
-    Equal, 
+    Equal,
 
     // open parentheses (
     OpenParen,
 
     // close parentheses )
-    CloseParen, 
+    CloseParen,
 
     // open bracket [
-    OpenBracket, 
-    
+    OpenBracket,
+
     // close bracket ]
     CloseBracket,
 
     // invalid token
-    Invalid(String)
+    Invalid(String),
 }
 
 #[derive(PartialEq, Clone, Debug)]
-pub struct Token{
-    pub token_type: TokenType
+pub struct Token {
+    pub token_type: TokenType,
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_eq_1(){
-        let t = Token{token_type: TokenType::Invalid("s".into())};
-        let r =  Token{token_type: TokenType::Invalid("s".into())};
-        assert_eq!(true, t == r);
-    }
-
-    #[test]
-    fn test_eq_2() {
-        let t = Token{token_type: TokenType::Plus};
-        let r = Token{token_type: TokenType::Plus};
-        assert_eq!(true, t == r);
-    }
-
-    #[test] 
-    fn test_eq_3() {
-        let t = Token{token_type: TokenType::Float(25.25)};
-        let r = Token{token_type: TokenType::Float(25.25)};
-        assert_eq!(true, t == r);
+impl fmt::Display for Token {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use TokenType::*;
+        write!(
+            f,
+            "{}",
+            match &self.token_type {
+                Float(num) => num.to_string(),
+                Int(num) => num.to_string(),
+                Plus => "+".into(),
+                Minus => "-".into(),
+                Mult => "*".into(),
+                Div => "/".into(),
+                Mod => "%".into(),
+                Exp => "^".into(),
+                Equal => "=".into(),
+                OpenParen => "(".into(),
+                CloseParen => ")".into(),
+                OpenBracket => "[".into(),
+                CloseBracket => "]".into(),
+                Invalid(s) => format!("Invalid({})", s),
+            }
+        )
     }
+}
 
```

Comment:
why do we use into for plus, minus, mult versus from. Since char and string are naturally converted. I may not understand the difference between from and into fully, but I think we should be differentiating.